xml2kinto/__main__.py

```diff
@@ -1,17 +1,92 @@
 
 # options to move to a config file
 xml_file = os.path.abspath(os.path.join(os.path.dirname(__file__),
                                         '..', 'blocklist.xml'))
 auth = ('mark', 'p4ssw0rd')
-collection_permissions = {'read': ["system.Everyone"]}
-bucket_name = u'blocklists'
-collection_name = u'certificates'
+COLLECTION_PERMISSIONS = {'read': ["system.Everyone"]}
+cert_bucket = u'blocklists'
+cert_collection = u'certificates'
+gfx_bucket = u'blocklists'
+gfx_collection = u'gfx'
+addons_bucket = u'blocklists'
+addons_collection = u'addons'
+plugins_bucket = u'blocklists'
+plugins_collection = u'plugins'
 kinto_server = 'http://localhost:8888/v1'
-fields = ('subject', 'pubKeyHash', 'serialNumber', 'issuerName')
+
+cert_items_fields = ('serialNumber', 'issuerName')
+gfx_items_fields = ('blockID', 'os', 'vendor', 'feature', 'featureStatus',
+                    'driverVersion', 'driverVersionComparator',
+                    ('devices', {'xpath': 'devices/*'}))
+
+addons_items_fields = (
+    'blockID',
+    ('id', {'name': 'guid'}),
+    ('prefs', {'xpath': 'prefs/*'}),
+    ('versionRange', {
+        'xpath': 'versionRange',
+        'fields': (
+            'minVersion',
+            'maxVersion',
+            'severity',
+            ('targetApplication', {
+                'xpath': 'targetApplication',
+                'fields': (
+                    ('id', {'name': 'guid'}),
+                    ('versionRange/minVersion', {'name': 'minVersion'}),
+                    ('versionRange/maxVersion', {'name': 'maxVersion'})
+                )
+            })
+        )
+    })
+)
+
+plugins_items_fields = (
+    'blockID', 'os',
+    ("match[@name='name']/exp", {'name': 'matchName'}),
+    ("match[@name='description']/exp", {'name': 'matchDescription'}),
+    ("match[@name='filename']/exp", {'name': 'matchFilename'}),
+    'infoURL',
+    ('versionRange', {
+        'xpath': 'versionRange',
+        'fields': (
+            'minVersion',
+            'maxVersion',
+            'severity',
+            ('vulnerabilitystatus', {'name': 'vulnerabilityStatus'}),
+            ('targetApplication', {
+                'xpath': 'targetApplication',
+                'fields': (
+                    ('id', {'name': 'guid'}),
+                    ('versionRange/minVersion', {'name': 'minVersion'}),
+                    ('versionRange/maxVersion', {'name': 'maxVersion'})
+                )
+            })
+        )})
+)
+
+
+def sync_records(fields, filename, xpath, kinto_client, bucket, collection):
+    xml_records = get_xml_records(
+        fields=fields,
+        filename=filename,
+        xpath=xpath)
+    kinto_records = get_kinto_records(
+        kinto_client=kinto_client,
+        bucket=bucket,
+        collection=collection,
+        permissions=COLLECTION_PERMISSIONS)
+
+    diff = get_diff(xml_records, kinto_records)
+    synchronize(diff, kinto_client,
+                bucket=bucket, collection=collection)
```

Comment:
If `synchronize` is not in charge of building the `diff`, maybe it should be named `push records` ?